pkg/apis/core/types.go

```diff
@@ -4142,11 +4142,48 @@ type NodeAddressType string
 
 // These are valid values of node address type
 const (
-	NodeHostName    NodeAddressType = "Hostname"
-	NodeExternalIP  NodeAddressType = "ExternalIP"
-	NodeInternalIP  NodeAddressType = "InternalIP"
-	NodeExternalDNS NodeAddressType = "ExternalDNS"
+	// NodeHostName identifies the node's canonical hostname
+	NodeHostName NodeAddressType = "Hostname"
+
+	// NodeInternalIP identifies an IP address which is assigned to one of the node's
+	// network interfaces. Every node should have at least one address of this type.
+	//
+	// An internal IP is normally expected to be reachable from every other node, but
+	// may or may not be visible to hosts outside the cluster. By default it is
+	// assumed that kube-apiserver can reach node internal IPs, though it is possible
+	// to configure clusters where this is not the case.
```

Comment:
just curious, when is this not possible?

Reply:
Some people want their control plane and the nodes to be on isolated networks from each other. The simplest form of this is that the control plane can't reach the internal IPs but the nodes all have externalIPs that the control plane can reach and you pass `--kubelet-preferred-address-types ExternalIP` to `kube-apiserver`. The more complicated form of this is that you set of some sort of authenticated proxy between the control plane and the nodes, which I forget the details of.